### PR TITLE
refactor: remove trailing whitespace

### PR DESCRIPTION
**Pull request by Miyelsh**

### CODE REVIEW

File: src/nvim/testdir/test_display.vim [hunk 1]
```diff
@@ -302,7 +302,7 @@ func Test_eob_fillchars()
   set fillchars=eob:+
   redraw
   call assert_equal('+', Screenline(2))
-  set fillchars=eob:\ 
+  set fillchars=eob:\
```
Comment (zeertzjq) [inline src/nvim/testdir/test_display.vim:305]: keep

File: src/nvim/testdir/test_display.vim [hunk 2]
```diff
@@ -313,7 +313,7 @@ func Test_display_linebreak_breakat()
   new
   vert resize 25
   let _breakat = &breakat
-  setl signcolumn=yes linebreak breakat=) showbreak=+\ 
+  setl signcolumn=yes linebreak breakat=) showbreak=+\
```
Comment (zeertzjq) [inline src/nvim/testdir/test_display.vim:316]: keep

File: src/nvim/testdir/test_cindent.vim [hunk 3]
```diff
@@ -765,8 +765,8 @@ func Test_cindent_1()
   }
 
   Constructor::Constructor(int a,
-                           int b 
-                          )  : 
+                           int b
+                          )  :
```
Comment (zeertzjq) [inline src/nvim/testdir/test_cindent.vim:769]: keep

File: src/nvim/testdir/test_cindent.vim [hunk 4]
```diff
@@ -775,7 +775,7 @@ func Test_cindent_1()
   }
 
   Constructor::Constructor(int a,
-                           int b )  : 
+                           int b )  :
```
Comment (zeertzjq) [inline src/nvim/testdir/test_cindent.vim:778]: keep

File: src/nvim/testdir/test_cindent.vim [hunk 5]
```diff
@@ -800,9 +800,9 @@ func Test_cindent_1()
      int Test() { return FALSE; }
      int Test1() { return TRUE; }
 
-     CAbc(int a, int b )  : 
+     CAbc(int a, int b )  :
         BaseClass(a)
-     { 
+     {
```
Comment (zeertzjq) [inline src/nvim/testdir/test_cindent.vim:805]: keep

File: src/nvim/testdir/test_cindent.vim [hunk 6]
```diff
@@ -917,8 +917,8 @@ func Test_cindent_1()
   }
 
   Constructor::Constructor(int a,
-  		int b 
-  		)  : 
+  		int b
+  		)  :
```
Comment (zeertzjq) [inline src/nvim/testdir/test_cindent.vim:921]: keep

File: src/nvim/testdir/test_cindent.vim [hunk 7]
```diff
@@ -1742,8 +1742,8 @@ func Test_cindent_1()
...
```
Comment (zeertzjq) [inline src/nvim/testdir/test_cindent.vim:1746]: keep

File: src/nvim/testdir/test_cindent.vim [hunk 8]
```diff
@@ -1752,7 +1752,7 @@ func Test_cindent_1()
   }
 
   Constructor::Constructor(int a,
-  		int b )  : 
+  		int b )  :
```
Comment (zeertzjq) [inline src/nvim/testdir/test_cindent.vim:1755]: keep

File: src/nvim/testdir/test_cindent.vim [hunk 9]
```diff
@@ -1777,9 +1777,9 @@ func Test_cindent_1()
   	int Test() { return FALSE; }
   	int Test1() { return TRUE; }
 
-  	CAbc(int a, int b )  : 
+  	CAbc(int a, int b )  :
   		BaseClass(a)
-  	{ 
+  	{
```
Comment (zeertzjq) [inline src/nvim/testdir/test_cindent.vim:1782]: keep

File: src/nvim/testdir/test_cindent.vim [hunk 10]
```diff
@@ -1894,8 +1894,8 @@ func Test_cindent_1()
   }
 
   Constructor::Constructor(int a,
-  		int b 
-  		)  : 
+  		int b
+  		)  :
```
Comment (zeertzjq) [inline src/nvim/testdir/test_cindent.vim:1898]: keep

File: src/nvim/testdir/test_cindent.vim [hunk 11]
```diff
@@ -2094,7 +2094,7 @@ func Test_cindent_2()
 
   let code =<< trim [CODE]
     {
-  
+
```
Comment (zeertzjq) [inline src/nvim/testdir/test_cindent.vim:2097]: keep

File: src/nvim/testdir/test_cindent.vim [hunk 12]
```diff
@@ -3485,7 +3485,7 @@ func Test_cindent_34()
   normal =][
 
   let expected =<< trim [CODE]
-  
+
```
Comment (zeertzjq) [inline src/nvim/testdir/test_cindent.vim:3488]: keep

File: src/nvim/testdir/test_fold.vim [hunk 13]
```diff
@@ -524,7 +524,7 @@ func Test_fold_create_marker_in_C()
   let content =<< trim [CODE]
     /*
      * comment
-     * 
+     *
```
Comment (zeertzjq) [inline src/nvim/testdir/test_fold.vim:527]: keep

File: src/nvim/testdir/test_join.vim [hunk 14]
```diff
@@ -420,7 +420,7 @@ int i = 7 /* foo *// 3
      ;
     ># Note that the last character of the ending comment leader (left angle bracket) is a comment leader itself. Make sure that this comment leader is not removed from the next line #< < On this line a new comment is opened which spans 2 lines. This comment should retain its comment leader.
     ># Note that the last character of the ending comment leader (left angle bracket) is a comment leader itself. Make sure that this comment leader is not removed from the next line #< < On this line a new comment is opened which spans 2 lines. This comment should retain its comment leader.
-    
+
```
Comment (zeertzjq) [inline src/nvim/testdir/test_join.vim:423]: keep

File: src/nvim/testdir/test_join.vim [hunk 15]
```diff
@@ -65,27 +65,27 @@ func Test_join_spaces_marks()
...
-asdfasdf 
+asdfasdf
 asdf
-asdfasdf  
+asdfasdf
 asdf
 asdfasdf	
 asdf
-asdfasdf	 
+asdfasdf
```
Comment (zeertzjq) [inline src/nvim/testdir/test_join.vim:88]: keep

**Review by zeertzjq — DISMISSED** [review]
From a rough skim these whitespace must be kept. There may be more. I recommend not touching any `src/nvim/testdir/test_*.vim` files.

**Comment by Miyelsh** [comment]
> From a rough skim these whitespace must be kept. There may be more. I recommend not touching any `*.vim` files.

Apologies for that. I agree, I will not touch any .vim files.

**Review by zeertzjq — COMMENTED** [review]
The changes to `regexp_nfa.c` look fine. Not sure about others as they are third-party libraries.

**Pull request closed without merging**